### PR TITLE
Session management

### PR DESCRIPTION
**Pull request by joacomiguens**
## Summary

- Handle user session management.

- Create navbar component.

- Create Home feature module.

- New routes pack set.

## Screenshot

session-management

## Trello Card

You can find it [here](https://trello.com/c/rotiX8SQ/16-session-management-logout)

### CODE REVIEW

File: wolox-books/src/app/app.component.html [hunk 1]
```diff
@@ -1 +1,2 @@
+<app-navbar></app-navbar>
```
Comment (ebedoyaalzate) [inline wolox-books/src/app/app.component.html]: Add this navbar just in home page

File: wolox-books/src/app/components/navbar/navbar.component.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<header class="nav">
+  <nav class="nav__content">
+    <img src="assets/LogoWolox.png" alt="Wolox logo" class="nav__content__logo" />
+    <a [routerLink]="['/login']" class="nav__content__auth" *ngIf="!(isLoggedIn | async)">Login</a>
```
Comment (ebedoyaalzate) [inline wolox-books/src/app/components/navbar/navbar.component.html]: This link it's not necessary

File: wolox-books/src/app/components/navbar/navbar.component.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+@Component({
+  selector: 'app-navbar',
+  templateUrl: './navbar.component.html',
+  styleUrls: ['./navbar.component.scss'],
+})
+export class NavbarComponent implements OnInit {
+  isLoggedIn: Observable<boolean>;
+
+  constructor(public userService: UserService) {
+    this.isLoggedIn = userService.isLoggedIn();
```
Comment (ebedoyaalzate) [inline wolox-books/src/app/components/navbar/navbar.component.ts]: Put the logic of the component in ngOnInit method

File: wolox-books/src/app/services/user.service.ts [hunk 4]
```diff
@@ -10,20 +10,47 @@ import { environment } from 'src/environments/environment';
 })
 export class UserService {
   private readonly API_URI: string = `${environment.apiUri}`;
+  private isLoginSubject: BehaviorSubject<boolean>;
 
-  constructor(private readonly http: HttpClient) {}
+  constructor(private readonly http: HttpClient) {
+    this.isLoginSubject = new BehaviorSubject<boolean>(this.hasToken());
```
Comment (ebedoyaalzate) [inline wolox-books/src/app/services/user.service.ts]: Add this initialization in the declaration

**Review by ebedoyaalzate — CHANGES_REQUESTED** [review]

File: wolox-books/src/app/app.component.html (hunk 1 above)
Comment (joacomiguens) [reply wolox-books/src/app/app.component.html]: Done 😄

File: wolox-books/src/app/components/navbar/navbar.component.html (hunk 2 above)
Comment (joacomiguens) [reply wolox-books/src/app/components/navbar/navbar.component.html]: Done 😄

File: wolox-books/src/app/components/navbar/navbar.component.ts (hunk 3 above)
Comment (joacomiguens) [reply wolox-books/src/app/components/navbar/navbar.component.ts]: Done 😄

File: wolox-books/src/app/services/user.service.ts (hunk 4 above)
Comment (joacomiguens) [reply wolox-books/src/app/services/user.service.ts]: Done 😄

File: wolox-books/src/app/screens/auth/screens/login/login.component.ts [hunk 5]
```diff
@@ -41,6 +44,7 @@ export class LoginComponent implements OnInit {
     this.userService.login(this.credentials).subscribe((accessToken) => {
       console.log('El access-token es:', accessToken);
```
Comment (mcallegari10) [inline wolox-books/src/app/screens/auth/screens/login/login.component.ts]: Remove console log

File: wolox-books/src/app/screens/home/components/navbar/navbar.component.html [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<header class="nav">
+  <nav class="nav__content">
+    <img src="assets/LogoWolox.png" alt="Wolox logo" class="nav__content__logo" />
+    <a
+      [routerLink]="['/login']"
```
Comment (mcallegari10) [inline wolox-books/src/app/screens/home/components/navbar/navbar.component.html]: I think the redirect should be done in the logout method

File: wolox-books/src/app/screens/home/home.component.scss [hunk 7]
```diff
@@ -0,0 +1,52 @@
+@import '../../../scss/variables';
+@import '../../../scss/mixins';
+@import '../../../scss/global-classes';
+
+.book {
+  @include card-shadow;
+  display: flex;
+  flex-direction: column;
+  padding: 32px 32px 20px 32px;
```
Comment (mcallegari10) [inline wolox-books/src/app/screens/home/home.component.scss]: You can use a shorthand like `padding: 32px 32px 20px;`

File: wolox-books/src/app/screens/home/home.component.scss [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+  flex-direction: column;
+  padding: 32px 32px 20px 32px;
+  transition: all 0.2s ease-in-out;
+
+  &:hover {
+    transform: scale(1.1);
+  }
+
+  &-link {
+    max-width: calc(320px - 40px);
```
Comment (mcallegari10) [inline wolox-books/src/app/screens/home/home.component.scss]: What is that `320px`?

**Review by mcallegari10 — CHANGES_REQUESTED** [review]

File: wolox-books/src/app/screens/home/home.component.scss (hunk 7 above)
Comment (joacomiguens) [reply wolox-books/src/app/screens/home/home.component.scss]: Cool ☝️

File: wolox-books/src/app/screens/home/components/navbar/navbar.component.html [hunk 9]
```diff
@@ -0,0 +1,8 @@
+<header class="nav">
+  <nav class="nav__content">
+    <img src="assets/LogoWolox.png" alt="Wolox logo" class="nav__content__logo" />
+    <a class="nav__content__auth" (click)="userService.logout()" *ngIf="isLoggedIn | async"
+      >Logout</a
+    >
```
Comment (mcallegari10) [inline wolox-books/src/app/screens/home/components/navbar/navbar.component.html:6]: Fix indentation 😝

File: wolox-books/src/app/screens/home/home.component.scss (hunk 8 above)
Comment (joacomiguens) [reply wolox-books/src/app/screens/home/home.component.scss]: I change it to 280px 😄

**Review by ebedoyaalzate — APPROVED** [review]

**Review by mcallegari10 — APPROVED** [review]

**Pull request merged**